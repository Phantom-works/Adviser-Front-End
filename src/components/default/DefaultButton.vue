<template>
  <button :class="{ 'disable-button': disabled }" :style="{ background, color }">
    {{ text }}
  </button>
</template>

<script lang="ts" setup>
defineProps({
  text: {
    type: String
  },
  color: {
    type: String
  },
  background: {
    type: String
  },
  disabled: {
    type: Boolean
  }
})
</script>

<style scoped>
.disable-button {
  opacity: 0.5;
}

button {
  background-color: #3f8b72;
  border: none;
  border-radius: 10px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 10px 20px;
  margin-left: 10px;
}

button:hover {
  filter: brightness(85%);
}
</style>
