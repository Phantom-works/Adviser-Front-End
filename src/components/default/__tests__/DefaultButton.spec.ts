import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import DefaultButton from '../DefaultButton.vue'

describe('DefaultButton Snapshots', () => {
  // Arrange

  it('Snapshot text empty ', () => {
    //Arrange
    // Act
    const wrapper = mount(DefaultButton, {
      props: {}
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Snapshot text long ', () => {
    //Arrange
    // Act
    const wrapper = mount(DefaultButton, {
      props: {
        text: 'garyhdjfiqhbfy8djfiqnd89hxioboqsid'
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Snapshot text small ', () => {
    //Arrange
    // Act
    const wrapper = mount(DefaultButton, {
      props: {
        text: 'ha0s'
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Snapshot disable true ', () => {
    //Arrange
    // Act
    const wrapper = mount(DefaultButton, {
      props: {
        disabled: true
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Snapshot disable false ', () => {
    //Arrange
    // Act
    const wrapper = mount(DefaultButton, {
      props: {
        disabled: false
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
