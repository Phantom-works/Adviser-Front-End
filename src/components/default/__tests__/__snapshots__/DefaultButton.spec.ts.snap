// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DefaultButton Snapshots > Snapshot disable false  1`] = `"<button data-v-dec1f997=\\"\\" class=\\"\\"></button>"`;

exports[`DefaultButton Snapshots > Snapshot disable true  1`] = `"<button data-v-dec1f997=\\"\\" class=\\"disable-button\\"></button>"`;

exports[`DefaultButton Snapshots > Snapshot text empty  1`] = `"<button data-v-dec1f997=\\"\\" class=\\"\\"></button>"`;

exports[`DefaultButton Snapshots > Snapshot text long  1`] = `"<button data-v-dec1f997=\\"\\" class=\\"\\">garyhdjfiqhbfy8djfiqnd89hxioboqsid</button>"`;

exports[`DefaultButton Snapshots > Snapshot text small  1`] = `"<button data-v-dec1f997=\\"\\" class=\\"\\">ha0s</button>"`;
